File: go/problems/day24_test.go
package problems

import "testing"

var day24 Solvable[int] = Day24{}

func TestDay24_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day24_1")
	result := day24.Solve(input)

	var expected int = 2
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay24_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day24_2")
	result := day24.Solve(input)

	var expected int = 20336
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day24.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day24.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 24", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day24_1");
    const result = solve(input);
    assert.equal(result, 2);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day24_2");
    const result = solve(input);
    assert.equal(result, 20336);
  });
});
